snpeff/5.1/Dockerfile

```diff
@@ -0,0 +1,49 @@
+FROM ubuntu:focal as app
+
+ARG SNPEFF_VER="5.1"
+
+# Metadata
+LABEL base.image="ubuntu:focal"
+LABEL dockerfile.version="1"
+LABEL software="SnpEff & SnpSift"
+LABEL software.version="5.1"
+LABEL description.SnpEff="Genetic variant annotation and effect prediction toolbox."
+LABEL description.SnpSift="Used after SnpEff annotation to filter and manipulate annotated files."
+LABEL website="https://pcingola.github.io/SnpEff"
+LABEL license="https://github.com/pcingola/SnpEff/blob/master/LICENSE.md"
+LABEL maintainer="Tom Iverson"
+LABEL maintainer.email="tiverson@utah.gov"
+
+# Install open using apt 
+RUN apt-get update && apt-get install -y software-properties-common &&\
+  add-apt-repository -y ppa:openjdk-r/ppa &&\
+  apt-get update && apt-get install -y \
+  openjdk-11-jre \
+  wget \
+  unzip
+
+# Use wget to install SnpEff and SnpSift
+RUN wget http://sourceforge.net/projects/snpeff/files/snpEff_latest_core.zip && \
```

Comment:
Related: https://github.com/pcingola/SnpEff/issues/392

It may not be possible at the moment, but perhaps I am missing something

Reply:
additionally I noticed that the jar files downloaded via the link in the dockerfile downloads an older version of snpEff (or at least the `-version` flag prints an older version

```
$ wget http://sourceforge.net/projects/snpeff/files/snpEff_latest_core.zip

$ unzip snpEff_latest_core.zip

$ cd snpEff/

$ java -jar snpEff.jar -version
SnpEff  4.3t    2017-11-24

```

Reply:
Yeah, I've looked into this quite a bit because I initially wanted to download a tagged v5.1 but I can't find a v5.1 anywhere, only the "snpeff_latest_core" which apparently is v4.3t.

In fact, [https://pcingola.github.io/SnpEff/](https://pcingola.github.io/SnpEff/) is the only place I find the 5.1 version stated. Like you said, the jar files -version flag prints v4.3t and the snpEff.config states to check [http://snpeff.sourceforge.net/versions.txt](http://snpeff.sourceforge.net/versions.txt) for the latest version.
```SNPEFF	4.3	2019-01-26	http://sourceforge.net/projects/snpeff/files/snpEff_latest_core.zip ```

I was thinking that "snpEff_latest_core" is tagged to always be the latest version. Hopefully you get a reply to [pcingola/SnpEff#392](https://github.com/pcingola/SnpEff/issues/392). I've asked Pablo Cingolani to connect on LinkedIn so I can message him and ask.

Reply:
I think it might be easier to download from github and install from source. You would have to install ANT and MAVEN during the apt-get phase.
```
snpeff_ver=5.1d

# get the tar file for the desired version:
wget https://github.com/pcingola/SnpEff/archive/refs/tags/v${snpeff_ver}.tar.gz && \
# unzip the tar file
tar -xvf ${snpeff_ver}.tar.gz && \
# remove the tar file (since it's not needed anymore)
rm ${snpeff_ver}.tar.gz 
```
And then use maven (mvn) to install the other dependencies as described in https://pcingola.github.io/SnpEff/download/#building-from-the-source